runtime/src/main/java/org/corfudb/util/serializer/DynamicProtobufSerializer.java

```diff
@@ -90,31 +92,28 @@ public DynamicProtobufSerializer(CorfuRuntime corfuRuntime) {
         ISerializer protobufSerializer = createProtobufSerializer();
         Serializers.registerSerializer(protobufSerializer);
 
-        // Open the Registry Table.
-        CorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>> corfuTable = corfuRuntime.getObjectsView()
+        // Open the Protobuf Descriptor Table.
+        CorfuTable<ProtobufFileName, CorfuRecord<ProtobufFileDescriptor, TableMetadata>> descriptorTable = corfuRuntime.getObjectsView()
                 .build()
-                .setTypeToken(new TypeToken<CorfuTable<TableName, CorfuRecord<TableDescriptors, TableMetadata>>>() {
+                .setTypeToken(new TypeToken<CorfuTable<ProtobufFileName, CorfuRecord<ProtobufFileDescriptor, TableMetadata>>>() {
                 })
-                .setStreamName(getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE, REGISTRY_TABLE_NAME))
+                .setStreamName(getFullyQualifiedTableName(CORFU_SYSTEM_NAMESPACE, PROTOBUF_DESCRIPTOR_TABLE_NAME))
```

Comment:
This map won't be garbage collected, so you will have two copies of the descriptor map. Why do you need both, can't you just keep one?

Reply:
This was just an optimization to avoid going to the corfu table on every access, which seemed to slow down checkpointing for some reason. Putting a map to cache the corfu table sped up the checkpointing. What might be good is if the corfu table itself can be closed and garbage collected with a new api.